src/nodes/kvs/gossip.rs

```diff
@@ -101,24 +101,14 @@ impl KvsNode {
 
     /// Sends out gossip messages to relevant nodes and threads.
     async fn send_out_gossip(&mut self) -> eyre::Result<()> {
+        self.gossip_epoch += 1;
         let work_start = Instant::now();
 
         let mut addr_keyset_map: HashMap<String, HashSet<Key>> = HashMap::new();
-        for key in self.local_changeset.drain() {
+        for key in self.local_changeset.drain().collect::<Vec<Key>>() {
```

Comment:
To avoid the extra allocation, we could replace the `self.local_changeset` with an empty set:

```suggestion
        for key in std::mem::take(&mut self.local_changeset) {
```